File: game_modules/stellar_objects/planets.py
class TerrestrialBody:
	def __init__(self, planet_id, name, orbit_index, star_instance, galaxy, **kwargs):
		# Planet Location and General Information
		self.galaxy = galaxy
		self.name = name

		# Parents
		self.parent_star = star_instance

		self.ids = {
			'self': planet_id,

			'star': star_instance.ids['self']
		}

		self.orbit_index = orbit_index  # 0, 1, 2 ... n
		self.orbital_distance = 1.4  # Radius from the center.
		self.eccentricity = 0

		# Planet Type information
		self.planet_type = 'terrestrial'
		self.file_path = '/path/to/file.png'

		# Planet contents information
		self.colonies = {}  # A dict of {id's: instances} of planet objects.
		self.resources = {
			'water': {
				'amount': 999999999999,  # 999,999,999,999
				'availability': 1
			},
			'build_materials': {
				'amount': 999999999999,  # 999,999,999,999
				'availability': 1
			},
		}

		# **kwarg update
		self.__dict__.update(kwargs)

		star_instance.planets[self.ids['self']] = self

	def serialize(self):
		dictionary = dict(self.__dict__)
		dictionary['colonies'] = dict()
		del dictionary['galaxy']
		del dictionary['parent_star']
		return dictionary

File: static/js/classes/stellar_objects/planets.js
/**
 * Created by Tomca on 9/03/2017.
 */

function Planet_object (name, ids, orbit_index, orbital_distance, eccentricity, planet_type, file_path, colonies, resources, world) {
    this.world = world;

    this.name = name;
    this.ids = ids;

    this.orbit_index = orbit_index;
    this.orbital_distance = orbital_distance;
    this.eccentricity = eccentricity;

    this.planet_type = planet_type;
    this.file_path = file_path;

    this.colonies = colonies;
    this.resources = resources;

    var config = this.world.get_config();
    this.parent_star = config.stars[this.ids['star']];

    this.parent_star.planets[this.ids['self']] = this;
}

function _create_planet(planet_instance, world) {
    new Planet_object(
        planet_instance['name'],
        planet_instance.ids,
        planet_instance.orbit_index,
        planet_instance.orbital_distance,
        planet_instance.eccentricity,
        planet_instance.planet_type,
        planet_instance.file_path,
        {},
        planet_instance.resources,
        world
    )
}
